java/src/main/java/com/cybersource/ws/client/HttpClientConnection.java

```diff
@@ -231,6 +237,28 @@ public boolean retryMethod(
             return false;
         }
     }
+    
+	@Override
+	void logRequestHeaders() {
+		Header[] headers=postMethod.getRequestHeaders();
+		Map<String, String> headerMap = new HashMap<String, String>();
+		for(Header header:headers){
+			headerMap.put(header.getName(), header.getValue());
+		 }
+        	logger.log(Logger.LT_INFO, "Request Headers: " + headerMap);
+	    }
+	
+	@Override
+	public void logResponseHeaders() {
+		Header[] headers = postMethod.getResponseHeaders();
+		 Map<String, String> headerMap = new HashMap<String, String>();
+		 for(Header header: headers){
```

Comment:
same as prev comment.

Reply:
headers array has key and value pairs.If im printing just headers, then the object is getting printed. Removed the map and printed as ArrayList.